TODO/todo_app/models.py

```diff
@@ -0,0 +1,41 @@
+from django.utils import timezone
+
+from django.db import models
+from django.urls import reverse
+
+# to add a default due time while creating a todo item
+def one_week_hence():
+    return timezone.now() + timezone.timedelta(days=7)
+
+#defining title of the list
+class ToDoList(models.Model):
+    title = models.CharField(max_length=100, unique=True)
+
+    def get_absolute_url(self):
+        return reverse("list", args=[self.id])
+
+    def __str__(self):
+        return self.title
+
+#defining item attribute
+class ToDoItem(models.Model):
+    title = models.CharField(max_length=100)
+    description = models.TextField(null=True, blank=True)
+    created_date = models.DateTimeField(auto_now_add=True)
+    due_date = models.DateTimeField(default=one_week_hence)
+    # Foreign key: each to-do item must belong to a to-do list
+    # on_delete: deletes the all the to-do items associated with a to-do list
+    todo_list = models.ForeignKey(ToDoList, on_delete=models.CASCADE)
+
+
+    def get_absolute_url(self):
```

Comment:
Same
> What will this function do?

Reply:
get_absolute_url(): Tells the django how to calculate the canonical URL for an object, returns a string that can be used to refer to an object over the HTTP.
for example:
def get_absolute_url(self):
        return reverse("list", args=[self.id])
Here,
      this function returns the URL for the list item refering to the id of the object.